specs/PushNotifications/PushNotifications-spec.md

```diff
@@ -0,0 +1,455 @@
+partner center portal by [associating an app with the Windows Store](https://docs.microsoft.com/azure/notification-hubs/notification-hubs-windows-store-dotnet-get-started-wns-push-notification). As a result, non-store provisioned <br/>
+apps like Win32 cannot avail of this feature even if they want to.
+
+**Client Registration**: Unpackaged 3rd party applications do not have a way today to register with the <br/>
+client because the API is private. The only way to get an implicit client side registration is to package <br/>
+the app as an MSIX Desktop Bridge or SSP which may not satisfy 3rd party requirements.
+
+**Push Channels**: All the Push APIs are dependent on the caller app having an identity at the minimal. <br/>
+However, even packaged Win32 apps that have an identity cannot use our Public APIs to subscribe to Push <br/>
+Channels because they may not be store provisioned. Sideloaded UWPs that aren't store distributed have <br/>
+exactly the same problem.
+
+**Activation**: When the native OS client receives a Push payload over the WNS socket, it signals the Background <br/>
+Infrastructure to trigger a task and run some code on the app's behalf. This flow is undefined for Win32 apps <br/>
+that need to use Push today mainly because of the Registration and API limitations described above. 
+
+**Channel Request Issues**: The channel APIs exposed today can throw exceptions via their Projections in the case <br/>
+of errors. Some errors can be critical errors while others are retryable errors. The msdn guidance for retrying isn't <br/>
+very clear and has been the root cause of multiple app bugs in the past. We want to abstract away the details of <br/>
+channel retry operations for developers in reunion.
+
+# Description
+
+At a high level, we need to provide a way for all Win32 applications to subscribe to and receive Push notifications <br/>
+irrespective of their app type. This includes unpackaged apps and packaged win32 (MSIX Desktop Bridge, Sparse Signed Packages). <br/>
+Moreover, all Push scenarios should adhere to OS resource management policies like Power Saver, Network Attribution, <br/>
+Enterprise group policies etc. The Reunion Push component will abstract away the complexities of dealing with WNS <br/>
+channel registrations and Push related activations as much as possible freeing the developer to focus on other app <br/>
+related challenges. 
+
+We will prioritize the following feature set for Reunion:
+* Supporting raw push for Packaged Win32(MSIX Desktop Bridge)
+* Supporting raw push for Unpackaged Win32.
+* Supporting Visual Cloud Toasts for packaged Win32 apps.
+* Supporting Visual Cloud Toasts for unpackaged Win32 apps.
+
+A Portal Registration flow through [AAD (Azure Active Directory)](https://docs.microsoft.com/azure/active-directory/develop/quickstart-register-app) will also be defined that removes the dependency of the Push Flow with <br/>
```

Comment:
We probably should call out that RemoteId in this spec maps to the ApplicationId(ClientId) in app registration for added clarity.

Reply:
@nicuparente Can you please make a similar reference in your spec. I think its important to have something E2E in the AAD registration spec that shows how we can get the AppId and use that in the Reunion APIs.